ownCloudSDK/Logging/Writers/OCLogFileWriter.m

```diff
@@ -120,17 +175,85 @@ - (void)appendMessage:(NSString *)message
 	}
 }
 
-- (nullable NSError *)eraseOrTruncate
+- (NSArray<OCLogFileRecord*>*)logRecords
 {
 	NSError *error = nil;
 
-	if ([[NSFileManager defaultManager] fileExistsAtPath:self.logFileURL.path])
+	// Get contents of directory
+	NSString *directoryPath = [[OCAppIdentity.sharedAppIdentity appGroupLogsContainerURL] path];
+	NSArray *directoryContents = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:directoryPath
+																					 error:&error];
+
+	NSMutableArray<OCLogFileRecord*> *records = [NSMutableArray new];
+
+	// Create an array of log records
+	for (NSString *filename in directoryContents)
 	{
-		if (![[NSFileManager defaultManager] removeItemAtURL:self.logFileURL error:&error])
+		NSString *fullPath = [directoryPath stringByAppendingPathComponent:filename];
+		NSDictionary* attributes = [self _attributesForPath:fullPath];
+		if(attributes)
+		{
+			NSDate *creationDate = attributes[NSFileCreationDate];
+			int64_t fileSize = [attributes[NSFileSize] longLongValue];
+			OCLogFileRecord *record = [[OCLogFileRecord alloc] initWithName:filename
+															   creationDate:creationDate
```

Comment:
This entire method and `OCLogFileRecord ` could be greatly simplified:
- using `-[NSFileManager contentsOfDirectoryAtURL:includingPropertiesForkeys:options:error:]` to shallow-enumerate the log folder, you'd get an array of URLs *and* could already pre-fetch attributes like size and creation date in a single line.
- designing `OCLogFileRecord` around and initializing its instances with log file URLs, the class can derive its property values inside the inititializer (or lazily - in property accessors) for size and date through `-[NSURL getResourceValue:forKey:error:]`.

That would remove all the code currently used to put paths together (both here and in `-[OCLogFileRecord fullPath]`) and use the modern URL-based APIs instead.

Reply:
@felix-schwarz makes sense.. changed